contracts/libraries/ReceiverWeights.sol

```diff
@@ -97,6 +97,14 @@ library ReceiverWeightsImpl {
         if (next == ADDR_END) next = ADDR_ROOT;
     }
 
+    /// @notice Checks if the list is fully zeroed and takes no storage space.
+    /// It means that either it was never used or that
+    /// it's been pruned after removal of all the elements.
+    /// @return True if the list is zeroed
+    function isZeroed(ReceiverWeights storage self) internal view returns (bool) {
```

Comment:
It's a little different and I wanted to avoid confusion. The list can be empty and its iterator will yield nothing, but still hold some data. The real `isEmpty` would need to use the iterator, but that would be more expensive than this simplified `isZeroed`, which is enough for this use case.